src/hidimstat/knockoffs.py

```diff
@@ -1,109 +1,580 @@
-# -*- coding: utf-8 -*-
-# Authors: Binh Nguyen <tuan-binh.nguyen@inria.fr>
-"""
-Implementation of Model-X knockoffs inference procedure, introduced in
-Candes et. al. (2016) " Panning for Gold: Model-X Knockoffs for
-High-dimensional Controlled Variable Selection"
-<https://arxiv.org/abs/1610.02351>
-"""
 import numpy as np
 from sklearn.preprocessing import StandardScaler
+from sklearn.covariance import LedoitWolf
+from sklearn.linear_model import LassoCV
+from sklearn.model_selection import KFold
+from sklearn.utils import check_random_state
 from sklearn.utils.validation import check_memory
+from joblib import Parallel, delayed
 
-from .gaussian_knockoff import _estimate_distribution, gaussian_knockoff_generation
-from .stat_coef_diff import _coef_diff_threshold, stat_coef_diff
+from hidimstat.gaussian_knockoff import (
+    gaussian_knockoff_generation,
+    repeat_gaussian_knockoff_generation,
+)
+from hidimstat.utils import fdr_threshold, quantile_aggregation
+
+
+def preconfigure_estimator_LassoCV(estimator, X, X_tilde, y, n_alphas=10):
+    """
+    Configure the estimator for Model-X knockoffs.
+
+    This function sets up the regularization path for the Lasso estimator
+    based on the input data and the number of alphas to use. The regularization
+    path is defined by a sequence of alpha values, which control the amount
+    of shrinkage applied to the coefficient estimates.
+
+    Parameters
+    ----------
+    estimator : sklearn.linear_model._base.LinearModel
+        An instance of a linear model estimator from sklearn.linear_model.
+        This estimator will be used to fit the data and compute the test
+        statistics. In this case, it should be an instance of LassoCV.
```

Comment:
```suggestion
        statistics. In this case, it must be an instance of LassoCV.
```

Reply:
Just curious: why do we need the user to provide the estimator, then ?

Reply:
This function is to configure the estimator. 
From your remark, I changed the usage of the function. I think it should be clearer now.